src/main/minio.js

```diff
@@ -1865,23 +1857,16 @@ export class Client {
   // Returns a function that can be used for uploading objects.
   // If multipart === true, it returns function that is used to upload
   // a part of the multipart.
-  getUploader(bucketName, objectName, contentType, multipart) {
+  getUploader(bucketName, objectName, metaData, multipart) {
     if (!isValidBucketName(bucketName)) {
       throw new errors.InvalidBucketNameError('Invalid bucket name: ' + bucketName)
     }
     if (!isValidObjectName(objectName)) {
       throw new errors.InvalidObjectNameError(`Invalid object name: ${objectName}`)
     }
-    if (!isString(contentType)) {
-      throw new TypeError('contentType should be of type "string"')
-    }
     if (!isBoolean(multipart)) {
       throw new TypeError('multipart should be of type "boolean"')
     }
-    if (contentType === '') {
```

Comment:
Should this be checked for within the metadata? If contentType is not set, add a default value like here to it